src/lib/components/Swap/index.tsx

```diff
@@ -12,7 +14,38 @@ import Settings from './Settings'
 import SwapButton from './SwapButton'
 import Toolbar from './Toolbar'
 
-export default function Swap() {
+interface DefaultTokenAmount {
+  address?: string | { [chainId: number]: string }
+  amount?: number
+}
+
+interface SwapDefaults {
+  tokenList: string | TokenList['tokens']
+  input: DefaultTokenAmount
+  output: DefaultTokenAmount
+}
+
+const DEFAULT_INPUT: DefaultTokenAmount = { address: 'ETH' }
+const DEFAULT_OUTPUT: DefaultTokenAmount = {}
+
+function useSwapDefaults(defaults: Partial<SwapDefaults> = {}): SwapDefaults {
+  const tokenList = defaults.tokenList || 'https://gateway.ipfs.io/ipns/tokens.uniswap.org'
```

Comment:
noice

Reply:
Thoughts on adding a schema validation check?

Reply:
the passed token list (whether it’s an object literal or a url string) will go through ajv to be validated, that validation just doesn’t happen until `useTokenList`.